streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1459,13 +1467,14 @@ public boolean equals(final Object o) {
             final Source source = (Source) o;
             // omit successor to avoid infinite loops
             return name.equals(source.name)
-                && topics.equals(source.topics);
+                && topics.equals(source.topics)
+                && topicPattern.equals(source.topicPattern);
```

Comment:
NPE: ![image](https://user-images.githubusercontent.com/925755/55269396-d55f3480-5292-11e9-9c29-78c524d63c65.png)

I'm not using a topic pattern, equality should still work.

Reply:
cc @mjsax is this an overlooked issue?

Reply:
Seems like. Just double checked the code and `topicPattern` could be `null`. This bug was introduced in `2.1` release.

@unthingable Thanks for pointing this out. Would you like to do a PR (I create a JIRA for it: https://issues.apache.org/jira/browse/KAFKA-8240)